src/inmanta/agent/forking_executor.py

```diff
@@ -163,7 +169,40 @@ def connection_lost(self, exc: Exception | None) -> None:
 
 
 class ExecutorClient(FinalizingIPCClient[ExecutorContext], LogReceiver):
-    pass
+    def __init__(self, name: str):
+        super().__init__(name)
+
+        # Keeps track of when this client was active last
+        self.last_used_at: datetime.datetime = datetime.datetime.now().astimezone()
+
+    def get_idle_time(self) -> timedelta:
```

Comment:
IMO this should be the qualified `datetime.timedelta`. Not a strong requirement for me, I only mention it because we try to be more aware of our import conventions.